File: 190.reverse-bits.py
#
# @lc app=leetcode id=190 lang=python3
#
# [190] Reverse Bits
#

# Time : O(1)
# Space: O(1)

# @lc code=start
class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        for i in range(32):
            res = (res << 1) | (n & 1)
            n >>= 1
        return res


# @lc code=end

File: 190.reverse-bits.go
/*
 * @lc app=leetcode id=190 lang=golang
 *
 * [190] Reverse Bits
 */

package leetcode

// @lc code=start
func reverseBits(num uint32) uint32 {
	var res uint32
	for i := 0; i < 32; i++ {
		res = res<<1 | num&1
		num >>= 1
	}
	return res
}

// @lc code=end
